sat_vapor_pres/include/sat_vapor_pres_k.inc

```diff
@@ -226,40 +92,43 @@
    endif
 
       table_siz = table_size
-      dtres = (tcmax - tcmin)/real(table_size-1)
-      tminl = real(tcmin)+TFREEZE  ! minimum valid temp in table
-      dtinvl = 1./dtres
-      tepsl = .5*dtres
-      tinrc = .1*dtres
-      if(present(teps )) teps =tepsl
-      if(present(tmin )) tmin =tminl
-      if(present(dtinv)) dtinv=dtinvl
+      dtres  = (real(tcmax,r8_kind)-real(tcmin,r8_kind))/real(table_size-1,r8_kind)
+      tminl  = real(tcmin,r8_kind)+real(TFREEZE,r8_kind)  ! minimum valid temp in table
+      dtinvl = 1.0_r8_kind/dtres
+      tepsl  = 0.5_r8_kind*dtres
+      tinrc  = 0.1_r8_kind*dtres
+      if(present(teps )) teps =real(tepsl,  FMS_SVP_KIND_)
+      if(present(tmin )) tmin =real(tminl,  FMS_SVP_KIND_)
+      if(present(dtinv)) dtinv=real(dtinvl, FMS_SVP_KIND_)
 
 ! To be able to compute tables for any temperature range and resolution,
 ! and at the same time exactly reproduce answers from memphis revision,
 ! it is necessary to compute ftact differently than it is in memphis.
-      tfact = 5.0*dtinvl
+      tfact = 5.0_r8_kind*dtinvl
 
-      hdtinv = dtinvl*0.5
+      hdtinv = 0._r8_kind*dtinvl
 
 ! compute es tables from tcmin to tcmax
 ! estimate es derivative with small +/- difference
 
       if (do_simple) then
 
+        ! TABLE = 610.78ES0*exp(-HLV/RGAS[1/tem - 1.TFREEZE])
```

Comment:
This just seems to be a hanging equation.  I think you should note it describes the equations below.